crates/services/sync/src/import/back_pressure_tests.rs

```diff
@@ -0,0 +1,273 @@
+use std::time::Duration;
+
+use fuel_core_services::stream::BoxStream;
+use fuel_core_types::{
+    blockchain::primitives::BlockId,
+    fuel_tx::Transaction,
+};
+
+use crate::ports::{
+    BlockImporterPort,
+    MockBlockImporterPort,
+    MockConsensusPort,
+    MockPeerToPeerPort,
+};
+
+use super::{
+    tests::empty_header,
+    *,
+};
+use test_case::test_case;
+
+#[derive(Default)]
+struct Input {
+    headers: Duration,
+    transactions: Duration,
+    consensus: Duration,
+    executes: Duration,
+}
+
+#[test_case(
+    Input::default(), State::new(None, None),
+    Config{
+        max_get_header_requests: 1,
+        max_get_txns_requests: 1,
+    }
+    => Count::default() ; "Empty sanity test"
+)]
+#[test_case(
+    Input {
+        headers: Duration::from_millis(10),
+        ..Default::default()
+    },
+    State::new(None, 1),
+    Config{
+        max_get_header_requests: 1,
+        max_get_txns_requests: 1,
+    }
+    => is less_or_equal_than Count{ headers: 1, transactions: 1, consensus_calls: 1, executes: 1, blocks: 1 }
+    ; "Single with slow headers"
+)]
+#[test_case(
+    Input {
+        headers: Duration::from_millis(10),
+        ..Default::default()
+    },
+    State::new(None, 100),
+    Config{
+        max_get_header_requests: 10,
+        max_get_txns_requests: 10,
+    }
+    => is less_or_equal_than Count{ headers: 10, transactions: 10, consensus_calls: 10, executes: 1, blocks: 21 }
+    ; "100 headers with max 10 with slow headers"
+)]
+#[test_case(
+    Input {
+        transactions: Duration::from_millis(10),
+        ..Default::default()
+    },
+    State::new(None, 100),
+    Config{
+        max_get_header_requests: 10,
+        max_get_txns_requests: 10,
+    }
+    => is less_or_equal_than Count{ headers: 10, transactions: 10, consensus_calls: 10, executes: 1, blocks: 21 }
+    ; "100 headers with max 10 with slow transactions"
+)]
+#[test_case(
+    Input {
+        consensus: Duration::from_millis(10),
+        ..Default::default()
+    },
+    State::new(None, 100),
+    Config{
+        max_get_header_requests: 10,
+        max_get_txns_requests: 10,
+    }
+    => is less_or_equal_than Count{ headers: 10, transactions: 10, consensus_calls: 10, executes: 1, blocks: 21 }
+    ; "100 headers with max 10 with slow consensus"
+)]
+#[test_case(
+    Input {
+        executes: Duration::from_millis(10),
+        ..Default::default()
+    },
+    State::new(None, 50),
+    Config{
+        max_get_header_requests: 10,
+        max_get_txns_requests: 10,
+    }
+    => is less_or_equal_than Count{ headers: 10, transactions: 10, consensus_calls: 10, executes: 1, blocks: 21 }
```

Comment:
Hmm, how does comparation work? It will compare each field and only if all fields `<=` then `true`?

Reply:
I'm assuming it just uses the existing derivation of `Ord`/`PartialOrd` for the `Count` type

Reply:
Yeh exactly. It's just checking the tests never exceed those maximum values